homeassistant/components/meteo_france/weather.py

```diff
@@ -35,6 +35,7 @@ def format_condition(condition: str):
     for key, value in CONDITION_CLASSES.items():
         if condition in value:
             return key
+    _LOGGER.debug("'%s' not in CONDITION_CLASSES", condition)
```

Comment:
We don't want side effects in entity state properties.

Reply:
Hello Martin, I don't understand your comment. To explain you this piece of code. weeather condition is returned by the API with strings in french. In the `const.py`file we have created a translation matrix from API to recommended value described [here](https://developers.home-assistant.io/docs/core/entity/weather/#recommended-values-for-state-and-condition).
I have noticed that sometime new string values are added by the API (and we don't any more access to complete list of possible values). In this case the icon in weather card is blank.
This debug message is to allow the users who will notice this bug to alert us on the new value if they want.

Reply:
The `format_condition` function is used in entity state properties eg `forecast`. We don't allow side effects like logging in entity state properties. If you want to log that information it needs to be done in a method and not in the state properties.

Reply:
OK I've understood now. Thanks for the explanation. So I will remove it for this PR and will check in the future how to do it by extending the coordinator callback.
Should I remove [this](https://github.com/home-assistant/core/blob/a2ab0867e2306e6440a786303767472f8a0970a9/homeassistant/components/meteo_france/weather.py#L133) other logging in state property?

Reply:
Yes, please remove that logging too.

Reply:
Done